theurgy.lic

```diff
@@ -256,6 +251,7 @@ class TheurgyActions
   def holy_water?
     DRCI.get_item_safe(@water_holder, @theurgy_supply_container)
     DRCI.inside?('holy water', @water_holder)
+    DRCI.put_away_item?(@water_holder, @theurgy_supply_container)
```

Comment:
I think this changes the meaning of `holy_water?`. Instead of returning whether or not `holy water` was found inside `@water_holder`, doesn't it now return whether or not `@water_holder` could be returned to `@theurgy_supply_container`?

Reply:
It is.  By not returning it to the right container at the end of this action, it leaves it in-hand which later causes snags when a badly timed stow_hands uses `stow` which may or may not put it in the right place.  All related actions get the water holder from theurgy container (filling the container or using it) so this is the only place that randomly moves it.

Reply:
Sorry, let me clarify. I understand _why_ you added the cleanup here.

What I'm saying is that the return value of `holy_water?` has a dramatically different _meaning_ now than it did before. For example, if you have a water holder in your supply container, but it has no water in it, I think this function now returns `true` when it used to return `false`. I'm worried that this has ripple effects.

Reply:
Ahhhh  I caught up!  I see what you mean.

Reply:
```
  def holy_water?
    DRCI.get_item_safe(@water_holder, @theurgy_supply_container)
    result = DRCI.inside?('holy water', @water_holder)
    DRCI.put_away_item?(@water_holder, @theurgy_supply_container)
    result
  end
  ```
  This should work without adding more logic?

Reply:
Looks good